src/main/java/github/scarsz/discordsrv/modules/requirelink/RequireLinkModule.java

```diff
@@ -67,13 +67,27 @@ private void check(String eventType, EventPriority priority, String playerName,
                     return;
                 }
             }
-            if (Bukkit.getServer().getBannedPlayers().stream().anyMatch(p -> p.getUniqueId().equals(playerUuid))) {
-                DiscordSRV.debug("Player " + playerName + " is banned, skipping linked check");
-                return;
-            }
-            if (Bukkit.getServer().getIPBans().stream().anyMatch(ip::equals)) {
-                DiscordSRV.debug("Player " + playerName + " connecting with banned IP " + ip + ", skipping linked check");
-                return;
+            boolean onlyCheckBannedPlayers = onlyCheckBannedPlayers();
+            if (!checkBannedPlayers() || onlyCheckBannedPlayers) {
+                boolean banned = false;
+                if (Bukkit.getServer().getBannedPlayers().stream().anyMatch(p -> p.getUniqueId().equals(playerUuid))) {
+                    if (!onlyCheckBannedPlayers) {
+                        DiscordSRV.debug("Player " + playerName + " is banned, skipping linked check");
+                        return;
+                    }
+                    banned = true;
+                }
+                if (!banned && Bukkit.getServer().getIPBans().stream().anyMatch(ip::equals)) {
+                    if (!onlyCheckBannedPlayers) {
+                        DiscordSRV.debug("Player " + playerName + " connecting with banned IP " + ip + ", skipping linked check");
+                        return;
+                    }
+                    banned = true;
+                }
+                if (onlyCheckBannedPlayers && !banned) {
+                    DiscordSRV.debug("Player " + playerName + " is bypassing link requirement, player is not banned");
```

Comment:
Should say something like `(beacuse Only check banned players is enabled)`

Reply:
Done.